Document `IsImplicitlyDefined` for SDK-defined PackageReference items and CPM behavior (#3586)
NuGet/docs.microsoft.com-nuget: docs/consume-packages/Package-References-in-Project-Files.md

MSBuild properties and package identities do not have the same restrictions so the package identity needs to be changed to an MSBuild friendly name, prefixed by the word `Pkg`.
To verify the exact name of the property generated, look at the generated [nuget.g.props](../reference/msbuild-targets.md#restore-outputs) file.

## `IsImplicitlyDefined` metadata

Some SDKs add `PackageReference` items automatically.
If you author an SDK that defines a `PackageReference`, set `IsImplicitlyDefined="true"` on that item.
This metadata tells NuGet tooling to treat the package version as controlled by the SDK.
Visual Studio's NuGet package manager and `dotnet package update` won't offer package upgrades for implicitly defined packages.

```xml
<ItemGroup>
  <PackageReference Include="PackageName" Version="1.2.3" IsImplicitlyDefined="true" />
</ItemGroup>
```

> When a project uses [Central Package Management (CPM)](Central-Package-Management.md), don't define a matching `<PackageVersion />` item in `Directory.Packages.props` for an implicitly defined package.

Reviewer: ```suggestion
When a project uses [Central Package Management (CPM)](Central-Package-Management.md), a `<PackageVersion />` item must not be defined in `Directory.Packages.props` for an implicitly defined package.
```

It's varying between who it's talking to (SDK author or package author). 

I'd lean for the language being more matter of fact.

> For implicitly defined packages, the version must come from the `PackageReference` item itself.

Reviewer 2: ```suggestion
When a project uses [Central Package Management (CPM)](Central-Package-Management.md), a `<PackageVersion />` item must not be defined in `Directory.Packages.props` for an implicitly defined package and instead must come from the `PackageReference` item itself.
```

> For implicitly defined packages, the version must come from the `PackageReference` item itself.

Reviewer: nit: 246 and 247, can probably be merged in the same sentence.
